### ISSUE TITLE
UX: Display saving status in menu bar

### ISSUE THREAD

**Issue by kathymakes** [LLK/scratch-www#2275]
### Overview
We are in the process of thinking through the GUI feedback for saving, autosaving, and other stages of the project lifecycle. 

For the time being in this iteration, we are going to first focus on these cases:

___

### GUI Feedback for Saving
- includes autosave, manual save
- implement the bare bones "Saving" text on the menu bar
![image](https://user-images.githubusercontent.com/8203939/48229064-fe7b3900-e374-11e8-9a08-bdd237a6e888.png)

- display a faded "Saved" within the first few moments of saving (we can define timeframe later.)
![image](https://user-images.githubusercontent.com/8203939/48229148-3bdfc680-e375-11e8-86d7-edc5161de5d1.png)
- if can't save (because of network connectivity etc), and Scratcher tries to navigate away, abort navigation. Maybe throw a js browser alert.

Tracking more aspects of this UX in: https://github.com/LLK/scratch-gui/issues/3727
___

### GUI Feedback for Cloud Variables

We are separately tracking this in https://github.com/LLK/scratch-www/issues/2285

___
**More thoughts about this here:**
**[Google doc notes](https://docs.google.com/document/d/1tDLtzmKx1IGO1kkEk_qPNND1QPz2mrpEGlxjJPCJD3k/edit
)**

**Comment by kathymakes** [comment]
### TLDR: When should we use flash alerts?
Recently we had some discussion about flash alerts and when to use them.
This is an example of a flash alert.
<img width="256" alt="project saved" src="https://user-images.githubusercontent.com/8203939/47870107-1421b900-dddf-11e8-8d82-367e732dae09.png">

Flash alerts, we decided, fall into two categories:
- You perform an action, and flash alert tells you that Scratch is processing your request (e.g. saving)
- Scratch notices that something changed (e.g. lost connectivity) and gives you tools to fix it

For now, we are thinking to only use them for:
- **Extension disconnection**
- **Saving from the menu**
- **Downloading/uploading**

**Comment by kathymakes** [comment]
Changed the flow for "Saving a project" - after talking to @thisandagain and @carljbowman we decided to make the "Manual save" and "Auto save" a parallel experience, for the time being.

**Comment by kathymakes** [comment]
Also, here is a potential flow for save as a copy... might be too much?
![save copy 1 alt](https://user-images.githubusercontent.com/8203939/48075747-e5765a80-e1b1-11e8-8a35-e562ed8f5308.png)
![save copy 2 alt](https://user-images.githubusercontent.com/8203939/48075748-e5765a80-e1b1-11e8-8a54-9eade7ebe2d9.png)
![save copy 3 alt](https://user-images.githubusercontent.com/8203939/48075749-e5765a80-e1b1-11e8-8dc5-4fb91c4f1bf3.png)
![save copy 4 alt](https://user-images.githubusercontent.com/8203939/48075750-e5765a80-e1b1-11e8-948e-8158e589a137.png)

**Comment by carljbowman** [comment]
@kathymakes - Thanks for exploring this some more!

The "Save as Copy" flow above feels a bit much. Based on our conversations yesterday we should bias towards making flows, like Save, Save as Copy, Remix, as seamless as possible and not reload the page unless absolutely necessary. 

Regarding "Saving" UI in the toolbar: it might be nice to include a little spinner to show something is happening. 2.0 also reads out "Saved" once the process is complete. For the short term, we might want to mimic this as it can be used more generically for feedback, rather than more context specific feedback (e.g. "Project Copied"). 

That said, I am totally in favor of using the alert system for more specific feedback, but at this point it feels lower priority as we should get something more generic in for launch.

**Comment by kathymakes** [comment]
@carljbowman 

Yeah - I like those idea as well - spinner, extending the toolbar system, etc! Thanks for the thoughts 👍 

Links to screens in a (slightly) more parseable format: 
**[Invision](https://invis.io/6ZP038T4GNH)**

**Comment by benjiwheeler** [comment]
FYI, I implemented the spinner in gui in https://github.com/LLK/scratch-gui/pull/3655

**Comment by kathymakes** [comment]
@benjiwheeler @carljbowman 
Following our chat today, here is a quick recap:

### GUI Feedback for Saving
- includes autosave, manual save
- implement the bare bones "Saving" text on the menu bar
![image](https://user-images.githubusercontent.com/8203939/48229064-fe7b3900-e374-11e8-9a08-bdd237a6e888.png)

- display a faded "Saved" within the first few moments of saving (we can define timeframe later.)
![image](https://user-images.githubusercontent.com/8203939/48229148-3bdfc680-e375-11e8-86d7-edc5161de5d1.png)
- if can't save (because of network connectivity etc), and Scratcher tries to navigate away, abort navigation. Maybe throw a js browser alert.


### GUI Feedback for Cloud Variables
- implement message using alerts the first time someone uses cloud variables
![image](https://user-images.githubusercontent.com/8203939/48229389-ebb53400-e375-11e8-8976-b3d59b5e456f.png)
- More info here: https://github.com/LLK/scratch-www/issues/2285

**Comment by benjiwheeler** [comment]
Implemented in #3776.
